framework/src/main/java/org/checkerframework/common/reflection/DefaultReflectionResolver.java

```diff
@@ -487,47 +488,51 @@ private boolean isUnknownMethod(MethodInvocationTree tree) {
         Names names = Names.instance(context);
 
         List<Symbol> result = new ArrayList<>();
+        Method loadClass = null;
+        Symbol sym = null;
         try {
-            Method loadClass = Resolve.class.getDeclaredMethod("loadClass", Env.class, Name.class);
+            loadClass = Resolve.class.getDeclaredMethod("loadClass", Env.class, Name.class);
             loadClass.setAccessible(true);
-            Symbol sym = (Symbol) loadClass.invoke(resolve, env, names.fromString(className));
-            if (!sym.exists()) {
-                debugReflection("Unable to resolve class: " + className);
-                return Collections.emptyList();
-            }
-
-            ClassSymbol classSym = (ClassSymbol) sym;
-            while (classSym != null) {
-                for (Symbol s : classSym.getEnclosedElements()) {
-                    // check all member methods
-                    if (s.getKind() == ElementKind.METHOD) {
-                        // Check for method name and number of arguments
-                        if (names.fromString(methodName).equals(s.name)
-                                && ((MethodSymbol) s).getParameters().size() == paramLength) {
-                            result.add(s);
-                        }
-                    }
-                }
-                if (!result.isEmpty()) {
-                    break;
-                }
-                Type t = classSym.getSuperclass();
-                if (!t.hasTag(CLASS) || t.isErroneous()) {
-                    break;
-                }
-                classSym = (ClassSymbol) t.tsym;
-            }
-            if (result.isEmpty()) {
-                debugReflection("Unable to resolve method: " + className + "@" + methodName);
-            }
+        } catch (SecurityException | NoSuchMethodException | IllegalArgumentException e) {
+            ErrorReporter.errorAbort("Error in obtaining reflective method.");
+        }
+        try {
+            sym = (Symbol) loadClass.invoke(resolve, env, names.fromString(className));
         } catch (SecurityException
-                | NoSuchMethodException
                 | IllegalAccessException
                 | IllegalArgumentException
                 | InvocationTargetException e) {
-            debugReflection("Exception during resolution of reflective method: " + e.getMessage());
+            ErrorReporter.errorAbort("Error in invoking reflective method.");
+        }
```

Comment:
the loading logic up to this point for setting `sym` seems to be the same as below for setting `symClass`, can you extract the reflective loading logic and the error throwing into a common private helper method which returns a `Symbol`?